### PR TITLE
EVG-14633 Add logging to projects modified by variables/rotate endpoint

### PR DESCRIPTION
**Pull request by bynn**
[EVG-14633](https://jira.mongodb.org/browse/EVG-14633)

### Description 
add logging

### Testing 
added test to TestUpdateProjectVarsByValue

### CODE REVIEW

File: rest/data/project_test.go [hunk 1]
```diff
@@ -444,6 +444,7 @@ func (s *ProjectConnectorGetSuite) TestUpdateProjectVars() {
 
 func TestUpdateProjectVarsByValue(t *testing.T) {
 	require.NoError(t, db.ClearCollections(model.ProjectVarsCollection))
+	require.NoError(t, db.ClearCollections(event.AllLogCollection))
```
Comment (ablack12) [inline rest/data/project_test.go:447]: you can actually just add event.AllLogCollection to line 446

**Review by ablack12 — CHANGES_REQUESTED** [review]
Can you change the name to be a bit more informative about what kind of logging you've added?

**Comment by ablack12** [comment]
To fix the failing test, attach a dummy user to the context. (example: https://github.com/evergreen-ci/evergreen/blob/b05b458bc9a2922e78e82a1d37c722a553209635/rest/route/project_test.go#L52)

File: rest/data/project_test.go (hunk 1 above)
Comment (bynn) [reply rest/data/project_test.go]: done

**Review by ablack12 — APPROVED** [review]
🚀

**Pull request merged**